enhancements/kata-containers/kata-containers-dev-preview.md

```diff
@@ -0,0 +1,276 @@
+
+Via the CR it will be possible to select a subset of worker nodes.
+
+For deploying binaries on the host the current idea is to use a container image that can be mounted on the host level.  
+
+The goal is to be able to install the operator via OperatorHub.
+
+The Operator will:
+- Create a crio drop-in config file via machine config objects;
+- Automatically select the payload image that contains correct version of the kata binaries for the given version of the OpenShift;
+- Configure a `RuntimeClass` called `kata` which can be used to deploy workload that uses kata runtime;
+- Create a machine-config object to enable the `qemu-kiwi` RHCOS extension.
+
+The `RuntimeClass` and payload image names will be visible in the CR status.
+
+#### KataContainers Operator Goals
+- Create an API which supports installation of Kata Runtime on all or selected worker nodes
+- Configure CRI-O to use Kata Runtime on those worker nodes
+- Installation of the runtimeClass on the cluster, as well as of the required components for the runtime to be controlled by the orchestration layer.
+- Updates the Kata runtime
+- Uninstall Kata Runtime and reconfigure CRI-O to not use it.
+
+#### KataContainers Operator Non-Goals
+To keep the Kata Operator's goal to the lifecycle management of the Kata Runtime, it will only support installation configuration of the Kata Runtime. This operator will not interact with any runtime configuration, such as Pod Annoations supported by Kata.
+
+#### Proposal
+The following new API is proposed `kataconfiguration.openshift.io/v1`
+
+```go
+// Kataconfig is the Schema for the kataconfigs API
+type Kataconfig struct {
+  metav1.TypeMeta   `json:",inline"`
+  metav1.ObjectMeta `json:"metadata,omitempty"`
+
+  Spec   KataconfigSpec   `json:"spec,omitempty"`
+  Status KataconfigStatus `json:"status,omitempty"`
+}
+
+// KataconfigSpec defines the desired state of Kataconfig
+type KataconfigSpec struct {
+  // KataConfigPoolSelector is used to filer the worker nodes
+  // if not specified, all worker nodes are selected
+  // +optional
+  KataConfigPoolSelector *metav1.LabelSelector
+  
+  Config KataInstallConfig `json:"config"`
+}
+
+type KataInstallConfig struct {
+}
+
+// KataconfigStatus defines the observed state of KataConfig
+type KataconfigStatus struct {
+  // Name of the runtime class
+  runtimeClassName string `json:"runtimeClassName"`
+
+  // +required
+  kataImage string `json:"kataImage"`
+
+  // TotalNodesCounts is the total number of worker nodes targeted by this CR
+  TotalNodesCount int `json:"totalNodesCount"`
+
+  // CompletedNodesCounts is the number of nodes that have successfully completed the given operation
+  CompletedNodesCount int `json:"CompletedNodesCount"`
```

Comment:
So if we fully use the MCO, then all this stuff is covered by the MCO objects, specifically `machineconfigpool`.